### PR TITLE
Add support for wilba.tech wt65_h4

### PR DESCRIPTION
**Pull request by MrLys**
## Description

Add support for the h4 based off of h1. I have not been able to make via work after flashing this, as such via is not enabled. Usevia.app did work prior to flashing. `vendorId` and `productId` is based on [use-via/keyboards](https://github.com/the-via/keyboards/blob/master/v3/wilba_tech/wt65_h4/wt65_h4.json).

I'm unsure of who to put as maintainer in `keyboards/wilba_tech/wt65_h4/readme.md`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Added support for the h4 based off of h1.

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/wilba_tech/wt65_h4/config.h
Comment (waffle87) [inline keyboards/wilba_tech/wt65_h4/config.h:1]: This file should be removed.

File: keyboards/wilba_tech/wt65_h4/info.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        "cols": ["F5", "D5", "B1", "B2", "B3", "D3", "D2", "C7", "C6", "B6", "B5", "B4", "D7", "D6", "D4"],
+        "rows": ["F0", "F1", "F4", "F6", "F7"]
+    },
+    "processor": "atmega32u4",
+    "url": "https://wilba.tech",
+    "usb": {
+        "device_version": "0.0.1",
+        "pid": "0x0038",
+        "vid": "0x6582"
+    },
```
Comment (waffle87) [inline keyboards/wilba_tech/wt65_h4/info.json:17]:
```suggestion
    },
    "qmk": {
        "locking": {
            "enabled": true,
            "resync": true
        }
    },
```

File: keyboards/wilba_tech/wt65_h4/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                                     KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS),
+
+	// Fn3 Layer
+	[3] = LAYOUT_65_iso_blocker(
+		KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS,          KC_TRNS,
+		KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+		KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,                    KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS,                   KC_TRNS,   KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                                     KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS, KC_TRNS)
```
Comment (waffle87) [inline keyboards/wilba_tech/wt65_h4/keymaps/default/keymap.c:59]:
```suggestion
		KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      KC_RALT, KC_RGUI,        KC_LEFT, KC_DOWN, KC_RGHT)
```

File: keyboards/wilba_tech/wt65_h4/rules.mk [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
```
Comment (waffle87) [inline keyboards/wilba_tech/wt65_h4/rules.mk:12]:
```suggestion
# This file intentionally left blank
```

File: keyboards/wilba_tech/wt65_h4/info.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
+{
+    "manufacturer": "wilba.tech",
+    "keyboard_name": "WT65-H4",
+    "maintainer": "Wilba",
+    "bootloader": "atmel-dfu",
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "cols": ["F5", "D5", "B1", "B2", "B3", "D3", "D2", "C7", "C6", "B6", "B5", "B4", "D7", "D6", "D4"],
+        "rows": ["F0", "F1", "F4", "F6", "F7"]
+    },
```
Comment (waffle87) [inline keyboards/wilba_tech/wt65_h4/info.json:10]:
```suggestion
    },
    "features": {
        "bootmagic": true,
        "mousekey": true,
        "extrakey": true,
        "nrko": true
    },
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by waffle87** [comment]
Also, correct me if I'm wrong, but the only difference this keyboard has with `wilba_tech/wt65_h1` is the layout? If so, you should simply add the layout to that keyboard's `info.json`, and a keymap utilising that layout if you'd like.

**Comment by MrLys** [comment]
> Also, correct me if I'm wrong, but the only difference this keyboard has with `wilba_tech/wt65_h1` is the layout? If so, you should simply add the layout to that keyboard's `info.json`, and a keymap utilising that layout if you'd like.

Not entirely sure what the exact differences are. I just used it as a baseline. But I could just as well have used the wt65_g. The productid is also different, h4 has 0x0038.
If you think that is the best approach then I’ll do that.

**Comment by waffle87** [comment]
It appears these could probably all be consolidated into one. I'll have a look at it now and incorporate your layout here and reach out if you're able to test.

**Comment by waffle87** [comment]
Can you please try my [`refactor/wt65`](https://github.com/waffle87/qmk_firmware/tree/refactor/wt65/keyboards/wilba_tech/wt65_h) branch. You can check it out locally using:
```shell
git remote add waffle https://github.com/waffle87/qmk_firmware
git fetch waffle
git checkout refactor/wt65
qmk git-submodule
qmk clean
qmk flash -kb wilba_tech/wt65_h -km iso
```

**Comment by MrLys** [comment]
> Can you please try my [`refactor/wt65`](https://github.com/waffle87/qmk_firmware/tree/refactor/wt65/keyboards/wilba_tech/wt65_h) branch. You can check it out locally using:
> 
> ```shell
> git remote add waffle https://github.com/waffle87/qmk_firmware
> git fetch waffle
> git checkout refactor/wt65
> qmk git-submodule
> qmk clean
> qmk flash -kb wilba_tech/wt65_h -km iso
> ```
> 
> edit: The PR mentioned below contains the layout you are adding here, so that PR can be used instead.

Great! I’ll try to test it during the weekend. 
Any idea how to make it via compatible? Based on the documentation on the usevia website this should work, but after flashing it is not recognised.

File: keyboards/wilba_tech/wt65_h4/keymaps/default/keymap.c (hunk 2 above)
Comment (MrLys) [reply keyboards/wilba_tech/wt65_h4/keymaps/default/keymap.c:59]: Why should there only be one layer present in the default?
Can the boards have any number of layers (up to 15) given enough memory or are there other limiting factors?

**Comment by MrLys** [comment]
> > Great! I’ll try to test it during the weekend. Any idea how to make it via compatible? Based on the documentation on the usevia website this should work, but after flashing it is not recognised.
> 
> That requires via to have the proper json merged into their repo.

I see, that it what I thought. Shouldn’t [this](https://github.com/the-via/keyboards/tree/master/v3/wilba_tech/wt65_h4) be enough for that?

File: keyboards/wilba_tech/wt65_h4/keymaps/default/keymap.c (hunk 2 above)
Comment (drashna) [reply keyboards/wilba_tech/wt65_h4/keymaps/default/keymap.c:59]: Empty layers like this actually increase the firmware size.  Unpopulated layers aren't needed.  And for via itself, any non-populated layers will automatically be filled with transparencies.

Comment (MrLys) [reply keyboards/wilba_tech/wt65_h4/keymaps/default/keymap.c:59]: I see. I'll remove the empty layers

File: keyboards/wilba_tech/wt65_h4/rules.mk
Comment (drashna) [inline keyboards/wilba_tech/wt65_h4/rules.mk:1]: This should be removed.

File: keyboards/wilba_tech/wt65_h4/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	// Default layer
+	[0] = LAYOUT_65_iso_blocker(
+		QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,        KC_HOME,
+		KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_ENT,         KC_PGUP,
+		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_BSLS,                 KC_PGDN,
+		KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                 KC_UP,   KC_END,
+		KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      KC_RALT, KC_RGUI,        KC_LEFT, KC_DOWN, KC_RGHT)
+};
```
Comment (drashna) [inline keyboards/wilba_tech/wt65_h4/keymaps/default/keymap.c:36]:
```suggestion
    // Default layer
    [0] = LAYOUT_65_iso_blocker(
        QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,        KC_HOME,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_ENT,         KC_PGUP,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_BSLS,                 KC_PGDN,
        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                 KC_UP,   KC_END,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      KC_RALT, KC_RGUI,        KC_LEFT, KC_DOWN, KC_RGHT)
};
```

File: keyboards/wilba_tech/wt65_h4/info.json
Comment (drashna) [inline keyboards/wilba_tech/wt65_h4/info.json:1]: Please rename this file to `keyboard.json`

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**